lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -6,31 +6,261 @@
 #include "runtime/hostfunc.h"
 #include "runtime/instance/memory.h"
 
+#ifdef WASINN_BUILD_OPENVINO
+#include <inference_engine.hpp>
+#include <string>
+namespace IE = InferenceEngine;
+#endif
 namespace WasmEdge {
 namespace Host {
 
+const std::string map_target_to_string(uint32_t Target) {
+  switch (Target) {
+  case 0:
+    return "CPU";
+  case 1:
+    return "GPU";
+  case 2:
+    return "TPU";
+  default:
+    return "";
+  }
+}
+
 Expect<uint32_t> WasiNNLoad::body(Runtime::Instance::MemoryInstance *MemInst
                                   [[maybe_unused]],
                                   uint32_t BuilderPtr [[maybe_unused]],
                                   uint32_t BuilderLen [[maybe_unused]],
                                   uint32_t Encoding [[maybe_unused]],
                                   uint32_t Target [[maybe_unused]],
                                   uint32_t GraphPtr [[maybe_unused]]) {
-  return 0;
+  auto log = spdlog::get("WasiNN");
+  // GraphBuilders' Layout: |builder-0|builder-0 len|builder-1|builder-1 len|...
+  uint32_t *GraphBuilders;
+  uint32_t *Graph;
+  GraphBuilders = MemInst->getPointer<uint32_t *>(BuilderPtr, 1);
+  Graph = MemInst->getPointer<uint32_t *>(GraphPtr, 1);
+  if (Encoding == this->Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASINN_BUILD_OPENVINO
+    if (BuilderLen != 2) {
+      log->error("Wrong GraphBuilder Length {:d}, expecting 2", BuilderLen);
+      return -1;
+    }
+    IE::Core ie_;
+    std::string device_name = map_target_to_string(Target);
+    if (device_name.length() == 0) {
+      log->error("Device target {:d} not support!", Target);
+      return -1;
+    } else {
+      log->info("Using device: {:s}", device_name);
+    }
+    uint32_t xml_strings_len = GraphBuilders[1];
```

Comment:
The naming style: Not to use `_` in the variable name.
`XMLStringLen` or `XMLStrLen`